### ISSUE TITLE
Player.setSpeed makes Distortion

### ISSUE THREAD

**Issue by lkwWalter** [roymacdonald/ofxSoundObjects#40]
Hello,
Is there a way to use that Function in ofxSoundPlayerObject::setSpeed in a proper way? In my own app it leaded to distortion and also odd behavior, like stopping the playhead. I just used values between 0 and 2. Then I tried the example **ofxSoundPlayerObject** but there was the same problem..
Is there a solution to manipulate playback smoothly?
Thanks ahead!

**Comment by roymacdonald** [comment]
Hi,
I know this problem. It has to do with the ofSoundBuffer resampling implementation, which is part of the OF core not this addon.
check the experimental branch of this addon. 
Download also into your addons folder [ofxSamplerate](https://github.com/roymacdonald/ofxSampleRate) and also add it to your project using Project Generator.
This should allow you to change the playback speed with no problems.

Let me know if this works for you.

**Comment by lkwWalter** [comment]
Hey! Thank you it works just fine!
Here is the code. Its optimizable I guess.
`activeSample->sampledBuffer = player.getBuffer();
    samplerate.changeSpeed(player.getBuffer(),activeSample->sampledBuffer,speed, 0);
    playerSampled.load(activeSample->sampledBuffer, "sampled");
    player.stop();
    playerSampled.play();`

**Comment by roymacdonald** [comment]
Hi, in that case you can totally avoid using the experimental branch and just use master, but instead do `player.setSpeed(speed, -1, true);` which will preprocess the whole audio file you have loaded in the player.
It does essentially the same what you are doing but with a single command, and it is optimized as well. you can call this function as many times as you want to but be aware that if the audio file is too long it might take a bit to long to preprocess and you might get an audio buffer underflow
TL;DR
When you just call `setSpeed(speed)`, the last parameter is false, which means that it will not preprocess the audio file, but instead do it on the fly over each tiny chunk of data that goes into the output buffer, and it is in this case where the ofSoundBuffer::resample(...) fails and introduces that ugly distortion.
